c/parallel/parallel_for_ordered.cc

```diff
@@ -300,23 +300,29 @@ void ordered::parallel(function<void(size_t)> pre_ordered,
                        function<void(size_t)> post_ordered)
 {
   if (sch->n_threads <= 1) {
+    enable_monitor(true);
```

Comment:
What if there is an exception during the ordered loop execution -- will the monitor remain enabled, potentially later causing a crash in python?

Reply:
Yes, could be a problem. Also for `parallel_for_static()` when we don't start a parallel region. Can add `try`/`catch` around the `func(i)` call and stop monitor thread if there is an exception, but not sure what to do with the exception itself in this case.

Reply:
well, you can always just re-throw it

Reply:
Sounds like a plan :)

Reply:
I moved try-catch blocks out of the loop: seem more robust that way.

Reply:
Yes, though I'm not sure if anything besides the `*_ordered()` functions can throw. May still be a little bit better from the performance point of view.